### PR TITLE
Group playback rate

### PR DESCRIPTION
**Pull request by rjwright**
@shans @dstockwell finally time for a review on this one. Sorry it took so long.

### CODE REVIEW

File: src/group-constructors.js [hunk 1]
```diff
@@ -57,16 +57,33 @@
...
 
       if (!player.source)
         return;
+
       if (tf == null) {
         player._removePlayers();
         return;
       }
+
+      if (!timing) {
```
Comment (dstockwell) [inline src/group-constructors.js]: looks like this can move inside the branch below

File: src/group-constructors.js [hunk 2]
```diff
@@ -57,16 +57,33 @@
...
+        timing = shared.normalizeTimingInput(player.source.timing);
+      }
+      if (tf == 0 && player.playbackRate < 0) {
```
Comment (dstockwell) [inline src/group-constructors.js]: add a comment about what is happening here

File: src/player.js [hunk 3]
```diff
@@ -48,7 +48,11 @@
 
   scope.Player.prototype = {
     _ensureAlive: function() {
-      this._inEffect = this._source._update(this.currentTime);
+      if (this.playbackRate < 0 && this.currentTime === 0) {
+        this._inEffect = this._source._update(-1);
```
Comment (dstockwell) [inline src/player.js]: comment that this is dealing with end exclusivity

File: src/player.js [hunk 4]
```diff
@@ -98,8 +102,15 @@
       return this._playbackRate;
     },
     set playbackRate(value) {
+      if (value == this._playbackRate) {
+        return;
+      }
       var oldCurrentTime = this.currentTime;
       this._playbackRate = value;
+      this._startTime = null;
+      if (!this.paused) {
```
Comment (dstockwell) [inline src/player.js]: I don't think we should do this when we are idle? what is this.paused?

**Comment by rjwright** [comment]
@dstockwell looking at comments now.

I think we should land the rename before we land this. Then I will manually rename the files in this branch to match file names in the renamed branch, then rebase. Or can you think of a better way?

File: src/player.js (hunk 4 above)
Comment (rjwright) [inline src/player.js]: > I don't think we should do this when we are idle? what is this.paused?

Player.paused is an old concept. It should be replaced with playState == 'paused' but playState currently depends on it. I can remove it in a follow-up patch. I need to do a sweep of our whole API and make sure it's consistent with the spec.

I think you are correct: we shouldn't do this if we are idle.

**Pull request closed without merging**